### ISSUE TITLE
Color as type

### ISSUE THREAD

**Issue by r00tc0d3** [jwalton/gchalk#3]
Hi,

Is it an option to add Color as a type. Like

``` golang
type Color func(s ...string) string
```

So it can be used like.

``` golang
package main

import (
	"fmt"
	"github.com/jwalton/gchalk"
)

type Message struct {
	message string
	color gchalk.Color
}

func printMessage(m Message) {
	fmt.Println(m.color(m.message))
}

func main() {
	warning := Message{
		message: "Warning",
		color: gchalk.Yellow,
	}
	printMessage(warning)
	error := Message{
		message: "Error",
		color: gchalk.Red,
	}
	printMessage(error)
}
```

Else the user has to add the type itself. It's just convince but I think it is beter.

Regards, Marcel

**Comment by jwalton** [comment]
I can definitely add that.  Any objection to `ColorFn` instead of just `Color`?

**Comment by jwalton** [comment]
Released in v1.3.0.
